task on arrays/script.js

```diff
@@ -0,0 +1,104 @@
+function isArray(obj) {
+  return typeof obj ==="object" &&
+  Object.prototype.toString.call(obj) === "[object Array]";
+}
+function isArray2(obj) {
+  return !!obj && obj.constructor === Array;
+}
+function range(a, /*необязательный*/ b ,
+  /*необязательный*/ c) {
+  let first, last,
+    step = c || 1;
+  if (b) {
+    first = a;
+    last = b;
+  } else {
+    first = 0;
+    last = a;
+  }
+  let array = [];
+  for (let value = first; (step > 0) ? value < last :  value > last; value += step) {
+    array.push(value);
+  }
+  return array;
+}
+function compact(array) {
+  return array.filter((function(x) { return x }));
+}
+function compact2(array) {
+  let result = [];
+  for (var i = 0; i < array.length; i++) {
+    if (array[i]) result.push(array[i]);
+  }
+  return result;
+}
+function sum(array) {
+  return array.reduce((function(x, y) { return x + y }));
+}
+function sum2(array) {
+  let sum = 0;
+  for (var i = 0; i < array.length; i++) {
+    sum += array[i];
+  }
+  return sum;
+}
+function unique(array) {
+  return array.filter(function(value, index, array) {
+    return array.indexOf(value, index + 1) < 0 }
+  );
+}
+function last(array) {
+  return array[array.length - 1];
+}
+function excludeLast(array, delLength) {
+  delLength = delLength || 1;
+  if (delLength > 0) {
+    if (delLength < array.length )
+    array.length -= delLength;
+    else array.length = 0;
```

Comment:
initial array is mutated, -1
please use brackets for if else, it's better to read